geode-core/src/main/java/org/apache/geode/cache/client/internal/InvalidateOp.java

```diff
@@ -39,11 +47,38 @@
    * the server.
    *
    * @param pool the pool to use to communicate with the server.
-   * @param region the name of the region to do the entry keySet on
+   * @param regionName the name of the region to do the entry invalidate on.
+   * @param region the region to do the entry invalidate on.
+   * @param event the event for this invalidate operation
    */
-  public static void execute(ExecutablePool pool, String region, EntryEventImpl event) {
-    AbstractOp op = new InvalidateOpImpl(region, event);
-    pool.execute(op);
+  public static Object execute(ExecutablePool pool, String regionName, EntryEventImpl event,
+      boolean prSingleHopEnabled,
+      LocalRegion region) {
+    AbstractOp op = new InvalidateOpImpl(regionName, event, prSingleHopEnabled, region);
+
+    if (prSingleHopEnabled) {
+      ClientMetadataService cms = region.getCache().getClientMetadataService();
+      ServerLocation server =
+          cms.getBucketServerLocation(region, Operation.INVALIDATE, event.getKey(), null,
+              event.getCallbackArgument());
+      if (server != null) {
+        try {
+          PoolImpl poolImpl = (PoolImpl) pool;
+          boolean onlyUseExistingCnx = ((poolImpl.getMaxConnections() != -1
+              && poolImpl.getConnectionCount() >= poolImpl.getMaxConnections()) ? true : false);
```

Comment:
I think you can omit the `? true : false` as the preceding expression will already resolve to one or the other.

Reply:
Thanks for review.